Source/WebKit/UIProcess/Automation/libwpe/WebAutomationSessionWPE.cpp

```diff
@@ -454,15 +454,49 @@ void WebAutomationSession::platformSimulateKeyboardInteraction(WebPageProxy& pag
     }
 #endif // ENABLE(WPE_PLATFORM)
 }
+
+static bool isShiftedChar(UChar keyVal)
+{
+    // https://w3c.github.io/webdriver/#dfn-shifted-character
+    return (keyVal == '~' || keyVal == '|' || keyVal == '{' || keyVal == '}' || keyVal == '<' || keyVal == '>'
+        || keyVal == '!' || keyVal == '@' || keyVal == '#' || keyVal == '$' || keyVal == '%' || keyVal == '^'
+        || keyVal == '&' || keyVal == '*' || keyVal == '(' || keyVal == ')' || keyVal == '_' || keyVal == '+'
+        || (keyVal >= 'A' && keyVal <= 'Z'));
```

Comment:
We can optimize this a bit by first checking if keyVal > 128, then if it's between A and Z, and then using a switch instead, like the mac implementation does.